Networking/Repository/AbstractRepository.swift

```diff
@@ -225,17 +238,25 @@ fileprivate extension AbstractRepository {
     func perform(
         method: NetworkingMethod,
         path: String,
-        parameters: [String: Any]?,
-        headers: [String: String]?) -> SignalProducer<RawDataResponse, RepositoryError> {
+        parameters: [String: Any]? = .none,
+        headers: [String: String]? = .none,
+        requiresSession: Bool) -> SignalProducer<RawDataResponse, RepositoryError> {
         guard let url = buildURL(path: path) else { return SignalProducer(error: .invalidURL) }
         
+        var headers = headers
+        if requiresSession {
+            guard _sessionManager.isLoggedIn else { return SignalProducer(error: .unauthenticatedSession) }
+            headers = headers ?? [:]
+            headers![authenticationHeaders.first!.key] = authenticationHeaders.first!.value
+        }
```

Comment:
Sure, you mean create a `Dictionary` extension and add the `+` operator?

I'll add the comment.